include/highfive/bits/H5ReadWrite_misc.hpp

```diff
@@ -32,27 +33,36 @@ struct BufferInfo {
 // details implementation
 template <typename SrcStrT>
 struct string_type_checker {
-    static DataType getDataType(const DataType&, bool);
+    static DataType getDataType(const DataType&, const DataType&);
 };
 
 template <>
 struct string_type_checker<void> {
-inline static DataType getDataType(const DataType& element_type, bool) {
-    return element_type;
+inline static DataType getDataType(const DataType& element_type, const DataType& dtype) {
+    // TEMP. CHANGE: Ensure that the character set is properly configured to prevent converter
+    // issues on HDF5 <=v1.12.0
+    if (H5Tget_class(element_type.getId()) == H5T_STRING) {
```

Comment:
Just a comment, should we check whether the encoding is ASCII or UTF-8, and throw an exception otherwise? I wonder if someone could create a file with another encoding, and break other parts of the code with this change (i.e., we are no longer forcing UTF-8 as the memory type of the buffer).

Reply:
I would at a minimum link to your issue here, to better keep track of it.

Reply:
I added the check for ASCII files, otherwise I wouldn't feel comfortable forcing a character set that is wider than 8-bits. It might or might not happen, but just in case this is a safer fix for now.

I also added the comments you suggested, @matz-e!